core-services/chat/src/main/java/org/silverpeas/core/chat/servers/OpenfireServer.java

```diff
@@ -0,0 +1,178 @@
+
+/**
+ * <p>Openfire server management service</p>
+ * <p>It implements the access mechanism to the XMPP server OpenFire.</p>
+ * @author remipassmoilesel
+ */
+public class OpenfireServer implements ChatServer {
+
+  private SilverLogger logger = SilverLogger.getLogger(this);
+
+  /**
+   * URL target of the REST requests.
+   */
+  private final String url;
+
+  private final HttpRequester requester;
+
+  public OpenfireServer() {
+    SettingBundle settings = ChatServer.getChatSettings();
+    this.url = settings.getString("chat.xmpp.restUrl") + "/users";
+    String key = settings.getString("chat.xmpp.restKey");
+    this.requester = new HttpRequester(key);
+  }
+
+  @Override
+  public void createUser(final User user) throws ChatServerException {
+    // get cross domain login and password
+    ChatUser chatUser = ChatUser.fromUser(user);
+    Response response = null;
+    try {
+      HttpRequester requester = request(url);
+      response = requester.post(o -> {
+        o.put("username", chatUser.getChatLogin())
+            .put("password", chatUser.getChatPassword())
+            .put("name", chatUser.getDisplayedName());
+        if (StringUtil.isDefined(chatUser.geteMail())) {
+          o.put("email", chatUser.geteMail());
+        }
+        return o;
+      });
+
+      if (response.getStatus() != 201) {
+        logger.error("Error while creating XMPP user: {0}",
+            response.getStatusInfo().getReasonPhrase());
+        throw new ChatServerException(
+            SilverpeasExceptionMessages.failureOnAdding("XMPP user", chatUser.getId()));
+      }
+    } catch (Exception e) {
+      logger.error("Error while creating XMPP user: {0}", e.getMessage());
+      throw new ChatServerException(
+          SilverpeasExceptionMessages.failureOnAdding("XMPP user", chatUser.getId()), e);
+    } finally {
+      if (response != null) {
+        response.close();
+      }
+    }
+  }
+
+  @Override
+  public void deleteUser(final User user) throws ChatServerException {
+    ChatUser chatUser = ChatUser.fromUser(user);
+    Response response = null;
+    try {
+      HttpRequester requester = request(url, chatUser.getChatLogin());
+      response = requester.delete();
+
+      if (response.getStatus() != 200) {
+        logger.error("Error while deleting XMPP user: {0}",
+            response.getStatusInfo().getReasonPhrase());
+        throw new ChatServerException(
+            SilverpeasExceptionMessages.failureOnDeleting("XMPP user", chatUser.getId()));
+      }
+    } catch (Exception e) {
+      logger.error("Error while deleting XMPP user: {0}", e.getMessage());
+      throw new ChatServerException(
+          SilverpeasExceptionMessages.failureOnDeleting("XMPP user", chatUser.getId()), e);
+    } finally {
+      if (response != null) {
+        response.close();
+      }
+    }
+  }
+
+  @Override
+  public void createRelationShip(final User user1, final User user2) throws ChatServerException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.silverpeas.core.chat.ChatServerException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)